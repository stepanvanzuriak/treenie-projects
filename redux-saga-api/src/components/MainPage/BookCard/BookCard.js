import React from 'react'
import PropTypes from 'prop-types'
import { Card, CardTitle, CardText, Button } from 'reactstrap'

const BookCard = ({
  title,
  description,
  excerpt,
  publishDate,
  linkText,
  onButtonClick,
  id,
  ...rest
}) => (
  <Card body key={id} {...rest}>
    <CardTitle>
      <b>{title}</b>
    </CardTitle>
    <CardText>
      <b>Description:</b> {description}
    </CardText>
    <CardText>
      <b>Excerpt:</b> {excerpt}
    </CardText>
    <CardText>
      <b>PublishDate:</b> {publishDate}
    </CardText>
    <Button outline color="info" onClick={() => onButtonClick(id)}>
      {linkText}
    </Button>
  </Card>
)

BookCard.defaultProps = {
  title: '',
  description: '',
  linkText: '',
  onButtonClick: () => {}
}

BookCard.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  linkText: PropTypes.string,
  onButtonClick: PropTypes.func
}

export default BookCard
