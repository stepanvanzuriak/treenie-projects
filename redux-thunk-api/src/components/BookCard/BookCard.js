import React from 'react'
import PropTypes from 'prop-types'
import { Card, CardTitle, CardText } from 'reactstrap'

export const BookCard = ({
  title,
  description,
  excerpt,
  publishDate,
  onButtonClick,
  id,
  ...rest
}) => (
  <Card body key={id} {...rest}>
    <CardTitle>
      <b>{title}</b>
    </CardTitle>
    <CardText>
      <b>Description:</b> {description}
    </CardText>
    <CardText>
      <b>Excerpt:</b> {excerpt}
    </CardText>
    <CardText>
      <b>PublishDate:</b> {publishDate}
    </CardText>
  </Card>
)

BookCard.defaultProps = {
  title: '',
  description: '',
  excerpt: '',
  publishDate: '',
  id: 0,
  onButtonClick: () => {}
}

BookCard.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  excerpt: PropTypes.string,
  id: PropTypes.number,
  publishDate: PropTypes.string,
  onButtonClick: PropTypes.func
}
