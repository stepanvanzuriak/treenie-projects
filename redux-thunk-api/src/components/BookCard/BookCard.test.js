import React from 'react'
import renderer from 'react-test-renderer'
import { BookCard } from './BookCard'

it('renders correctly', () => {
  const tree = renderer.create(<BookCard />).toJSON()
  expect(tree).toMatchSnapshot()
})
