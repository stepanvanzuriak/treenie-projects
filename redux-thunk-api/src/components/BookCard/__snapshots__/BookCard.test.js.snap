// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card card-body"
>
  <h5
    className="card-title"
  >
    <b>
      
    </b>
  </h5>
  <p
    className="card-text"
  >
    <b>
      Description:
    </b>
     
    
  </p>
  <p
    className="card-text"
  >
    <b>
      Excerpt:
    </b>
     
  </p>
  <p
    className="card-text"
  >
    <b>
      PublishDate:
    </b>
     
  </p>
</div>
`;
